File: Data Structures & Algorithms/three-integer-sum/submission-3.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        res = []
        nums.sort()
        for i in range(len(nums)):
            
            if i > 0 and nums[i-1] == nums[i]:
                continue
            j = len(nums) - 1
            k = i + 1
            while k < j:
                if nums[i] + nums[k] + nums[j] == 0:
                    res.append([nums[i],nums[k],nums[j]])
                    k += 1
                    j -= 1
                    while k < j and nums[k] == nums[k-1]:
                        k += 1
                    while k < j and nums[j] == nums[j+1]:
                        j -= 1

                elif nums[i] + nums[k] + nums[j] > 0:
                    j -= 1
                elif nums[i] + nums[k] + nums[j] < 0:
                    k += 1
        return res

File: Data Structures & Algorithms/palindromic-substrings/submission-3.cpp
class Solution {
public:
    int countSubstrings(string s) {
        int res = 0;
        int size = s.size();
        int l = 0;
        int r = 0;

        for(int i = 0; i < size; i++)
        {
            l = i, r = i;
            while(l >= 0 && r < size && s[l--] == s[r++])
                res++;
            
            l = i, r = i + 1;
            while(l >= 0 && r < size && s[l--] == s[r++])
                res++;
        }

        return res;
    
    }

};
